addons/mail/models/mail_channel.py

```diff
@@ -318,6 +319,8 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                         group_name=channel.group_public_id.name,
                         partner_names=', '.join(partner.name for partner in invalid_partners)
                     ))
+                if not channel._can_invite_guest(guests=guests):
+                    raise werkzeug.exceptions.NotFound()
```

Comment:
we should move the check to the create of `channel.partner` like the other `_can_invite`, also it should not return an http error but a standard odoo error from inside models, because they can live without being used from a request.

Reply:
The method create of `channel.partner` is called when the channel is created and at that time guests are not created so we won't get guests there.

Reply:
@seb-odoo Could you please look into this? Thanks.

Reply:
Hello @ibo-odoo,
Sorry for the delay, the scope and size of the team grew too much and I had trouble following up everything, but @oomsveta should help me with reviews now, just give us a bit more time to train him to review and to catch up with the backblog. I'll make sure we get moving all the pending PR and tasks on our project in the following weeks. Thanks!